docs/concepts/configuration/assign-pod-node.md

```diff
@@ -190,13 +190,7 @@ flavor and the `preferredDuringSchedulingIgnoredDuringExecution` flavor.
 
 As with node affinity, the legal operators for pod affinity and anti-affinity are `In`, `NotIn`, `Exists`, `DoesNotExist`, `Gt`, `Lt`.
 
-In principle, the `topologyKey` can be any legal label value. However,
-for performance reasons, only a limited set of topology keys are allowed.
-By default the allowed topology keys are:
-
-* `kubernetes.io/hostname`
-* `failure-domain.beta.kubernetes.io/zone`
-* `failure-domain.beta.kubernetes.io/region`
+The `topologyKey` can be any legal label value.
```

Comment:
@gyliu513 Oh yeah, thanks for dive deep into the code, the failureDomains only affect the case when `topologyKey` was not indicated, which semantically means "all topologies".
(There was once an idea of optimizing the algorithm by indicating available topologyKeys but never got implemented.)

Anyway, just to conclude here:
1. For hard pod anti-affinity, without the admission controller "LimitPodHardAntiAffinityTopology" enabled, topologyKey can be any non-empty value. 
2. For soft pod anti-affiniy, topologyKey can be any value when indicated, but when not indicated, the semantic "all topologies" here only means kubeletapis.DefaultFailureDomains